scripts/mktcli.js

```diff
@@ -161,27 +151,14 @@ const main = {
 
           // publication[1] === owner. See `toPublicationLog`
           if (Contract.isEmptyAddress(publication[1].toString())) {
-            const landRegistry = eth.getContract('LANDRegistry')
-            const estateRegistry = eth.getContract('EstateRegistry')
+            const registryContract = getContractByAssetType(assetType)
 
             contract = eth.getContract('Marketplace')
 
-            switch (assetType) {
-              case ASSET_TYPES.parcel: {
-                publication = await contract.orderByAssetId(
-                  landRegistry.address,
-                  asset.token_id
-                )
-                break
-              }
-              case ASSET_TYPES.estate: {
-                publication = await contract.orderByAssetId(
-                  estateRegistry.address,
-                  asset.token_id
-                )
-                break
-              }
-            }
+            publication = await contract.orderByAssetId(
```

Comment:
What do you think of using the [legacy flag](https://github.com/decentraland/marketplace/pull/588/commits/192b97b43ed1421eac004210e3445f232f71dd65#diff-14c145377d6748040b73ee576c8e1d57R192) to get legacy active publication

Reply:
In this particular case, because it's a read, the command is checking for the legacy publication (`Contract.isEmptyAddress(publication[1].toString())`) and defaulting to the new one if it doesn't exist

We could have a problem if the asset has a publication in both contracts (unlikely but possible). But I think it might be better to just return everything instead of using a flag here.

Something like:

```
log.info(`(publication) id:(${asset.id})`)
log.info(`blockchain legacy => ${legacyPublication}`)
log.info(`blockchain        => ${publication}`)
log.info(`db                => ${publicationDb}`)
```

What do you think?

Reply:
As we are going to shut down legacy publications, it is going to be weird to have always the legacy log there. I still prefer to have the flag to decide when send the command in which marketplace I want to see the publication

Reply:
Hm but when we remove legacy publications this and the flag will be removed (both things).
Contrary to publication-cancel, I'm we're not making a destructive action which requires gas to work, the user needs information about an asset. If we add the flag here (a read) we're effectively making the user run it twice, once per each marketplace.
I want to know the publications associated with this asset, why should I know where that publication is?

I don't know if I'm really explaining this correctly

Reply:
Ahhh ok, I get the point. Agree with you now!

Reply:
Updated!